superset-frontend/src/components/Tags/Tag.tsx

```diff
@@ -46,41 +47,39 @@ const Tag = ({
   const tagElem = (
     <>
       {editable ? (
-        <StyledTag
-          key={id}
-          closable={editable}
-          onClose={handleClose}
-          color="blue"
-        >
-          {isLongTag ? `${name.slice(0, 20)}...` : name}
-        </StyledTag>
+        <Tooltip title={toolTipTitle} key={toolTipTitle}>
+          <StyledTag
+            key={id}
+            closable={editable}
+            onClose={handleClose}
+            color="blue"
+          >
+            {isLongTag ? `${name.slice(0, 20)}...` : name}
+          </StyledTag>
+        </Tooltip>
       ) : (
-        <StyledTag role="link" key={id} onClick={onClick}>
-          {id ? (
-            <a
-              href={`/superset/all_entities/?id=${id}`}
-              target="_blank"
-              rel="noreferrer"
-            >
-              {isLongTag ? `${name.slice(0, 20)}...` : name}
-            </a>
-          ) : isLongTag ? (
-            `${name.slice(0, 20)}...`
-          ) : (
-            name
-          )}
-        </StyledTag>
+        <Tooltip title={toolTipTitle} key={toolTipTitle}>
+          <StyledTag role="link" key={id} onClick={onClick}>
+            {id ? (
+              <a
+                href={`/superset/all_entities/?id=${id}`}
+                target="_blank"
+                rel="noreferrer"
+              >
+                {isLongTag ? `${name.slice(0, 20)}...` : name}
+              </a>
+            ) : isLongTag ? (
+              `${name.slice(0, 20)}...`
+            ) : (
+              name
+            )}
+          </StyledTag>
+        </Tooltip>
       )}
     </>
   );
 
-  return isLongTag ? (
-    <Tooltip title={name} key={name}>
-      {tagElem}
-    </Tooltip>
-  ) : (
-    tagElem
-  );
+  return tagElem;
```

Comment:
Do we have a test asserting we are rendering the "short" version of the tag when needed?